cmd/config.go

```diff
@@ -48,13 +48,24 @@ func getMcConfigDir() (string, *probe.Error) {
 	if e != nil {
 		return "", probe.NewError(e)
 	}
-	configDir := filepath.Join(homeDir, defaultMCConfigDir())
-	return configDir, nil
+	if runtime.GOOS == "windows" {
+		return filepath.Join(homeDir, mcDirName()), nil
+	}
+	// Check if ~/.mc already exists, deprecated Sep 2021
+	oldMcConfigDir := filepath.Join(homeDir, mcDirName())
+	if st, err := os.Stat(oldMcConfigDir); err == nil && st.IsDir() {
+		return oldMcConfigDir, nil
+	}
+	// Construct mc config dir path according to https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.6.html
+	xdgPath := os.Getenv("XDG_CONFIG_HOME")
+	if xdgPath == "" {
+		xdgPath = filepath.Join(homeDir, ".config")
```

Comment:
Just return the previous path `~/.mc`. No need to have even more places to look.

Reply:
> Just return the previous path `~/.mc`. No need to have even more places to look.

I am following the specification here, if XDG_CONFIG_HOME is not defined, then you need to use ~/.config/mc/

Reply:
Using ~/.config/mc is the correct behavior in this case. The only time ~/.mc should be used is if the config file cannot be found at a better location but exists in ~/.mc, i.e. for installs which pre-date this change.

Reply:
I don't really care about the specification, I just don't want the config to potentially be in 3 different places. 2 is confusing enough.

Reply:
No one else really cares about the spec, either, but to do the right thing you need to set the fallback to ~/.config/mc for users who don't have the environment variable set. Most users do not have the environment variable set and the problem will not be resolved to their satisfaction without this logic.